spl/src/governance.rs

```diff
@@ -34,7 +34,10 @@ macro_rules! vote_weight_record {
                 &self,
                 writer: &mut W,
             ) -> std::result::Result<(), ProgramError> {
-                anchor_lang::AnchorSerialize::serialize(&self.0, writer)
+                let mut to_write = &mut self.0.clone();
+                //to_write.account_discriminator = *b"2ef99b4b";
+                to_write.account_discriminator = VoterWeightRecord::discriminator();
```

Comment:
Yeah, this is a bit unfortunate. I wonder if it would make sense to have an
```
impl VoterWeightRecord {
   pub fn init(&mut self) {
       // sets discriminator
   }
}
```
that we require to be called when the account is first inited. Similar to how `AccountLoader::load_init()` works.

Reply:
can the `init` macro codegen be customized, e.g, by implementing a trait?